File: Trunk/Client/Assets/Script/CommonScriptsCode/ResManager/UI/Middleware/Mid_base_control_panel.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class Mid_base_control_panel:IMiddleware
{
	public GameObject go;
	public ImageWidget control_img;

	public Mid_base_control_panel(GameObject go) 
	{
		this.go =  go;
		control_img =  go.transform.Find("control_img").GetComponent<ImageWidget>();
	}

	public void DelReference() 
	{
#if TOOL
#else
		if(go!=null) GameObject.Destroy(go);
#endif
	}

}

File: Trunk/Client/res/lua/base/mid/Mid_base_control_panel.lua
﻿local ImageWidget=CS.ImageWidget

Mid_base_control_panel={}
local this = Mid_base_control_panel

function this:new(gameObject)
	local o = { }
	setmetatable(o, self)
	self.__index = self
	o:init(gameObject)
	return o
end

this.go = nil
this.control_img=nil

function this:init(gameObject)
	self.go=gameObject
	self.control_img=self.go.transform:Find("control_img"):GetComponent(typeof(ImageWidget))
end

